src/microsoft-dotnet-interactive/src/kernel.ts

```diff
@@ -155,21 +158,23 @@ export class Kernel {
     }
 
     handleCommand(commandEnvelope: contracts.KernelCommandEnvelope): Promise<void> {
+        if (!commandEnvelope.routingSlip) {
+            commandEnvelope.routingSlip = [];
+        }
         return new Promise<void>(async (resolve, reject) => {
             let context = KernelInvocationContext.establish(commandEnvelope);
 
             const previoudHendlingKernel = context.handlingKernel;
             context.handlingKernel = this;
             let isRootCommand = areCommandsTheSame(context.commandEnvelope, commandEnvelope);
 
-            let eventSubscription: rxjs.Subscription | undefined = undefined;//?
+            let eventSubscription: rxjs.Subscription | undefined = undefined;
 
             if (isRootCommand) {
                 this.name;//?
                 Logger.default.info(`kernel ${this.name} of type ${KernelType[this.kernelType]} subscribing to context events`);
                 eventSubscription = context.kernelEvents.pipe(rxjs.map(e => {
-                    const message = `kernel ${this.name} of type ${KernelType[this.kernelType]} saw event ${e.eventType} with token ${e.command?.token}`;
-                    message;//?
+                    const message = `kernel ${this.name} of type ${KernelType[this.kernelType]} saw event ${e.eventType} with token ${e.command?.token}`; Logger.default.info(message);
```

Comment:
The logger call on the end of this line is redundant.

Reply:
thank you